bin/s2nc.c

```diff
@@ -369,6 +376,12 @@ int main(int argc, char *const *argv)
         case 'L':
             key_log_path = optarg;
             break;
+        case 'P':
+            psk_list = s2n_array_new(sizeof(struct s2n_array));
+            POSIX_ENSURE_REF(psk_list);
+            POSIX_GUARD_RESULT(s2n_array_init(psk_list, sizeof(struct s2n_psk)));
+            POSIX_GUARD(s2n_setup_external_psk(psk_list, optarg));
+            break;
```

Comment:
You should avoid calling internal methods if possible. s2n_array isn't public.

Why not just save the optarg and process it once later when you can just call s2n_connection_append_psk directly?

Reply:
I think we should do the setup earlier and not delay it as this is where we validate the input `psk_identity`, `psk_secret` and `psk_hmac` values. I could consider using `struct s2n_psk_parameters` which contains `struct s2n_array psk_list;` within it as an alternative.